dds/DCPS/TypeSupportImpl.cpp

```diff
@@ -61,8 +61,8 @@ void TypeSupportImpl::to_type_info(XTypes::TypeInformation& type_info) const
   } else {
     const TypeObject& minTypeObject = pos->second;
     type_info.minimal.typeid_with_size.type_id = minTypeId;
-    type_info.minimal.typeid_with_size.typeobject_serialized_size =
-      serialized_size(get_typeobject_encoding(), minTypeObject);
+    const size_t sz = serialized_size(get_typeobject_encoding(), minTypeObject);
+    type_info.minimal.typeid_with_size.typeobject_serialized_size = static_cast<unsigned>(sz);
```

Comment:
Is it possible that `sz` is greater than the maximum of `unsigned` and is truncated by the casting?

Reply:
That would be a massive type object.

Reply:
I was thinking about systems where `unsigned` is 16-bit but I guess OpenDDS will never run on such systems, so this should work.

Reply:
I forgot that the spec says `unsigned` can be 16 bits, no I don't think we're going to encounter that. If it was an option I'd prefer us to `uint32_t`.

Reply:
It's 32-bit on all supported platforms.  uint32_t doesn't exist on all supported platforms.

Reply:
"If it was an option"